Content.IntegrationTests/Tests/Roles/StartingGearStorageTests.cs

```diff
@@ -0,0 +1,100 @@
+using System.Linq;
+using System.Numerics;
+using Content.Shared.Roles;
+using Content.Shared.Storage;
+using Content.Server.Storage.EntitySystems;
+using Robust.Shared;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Map;
+using Robust.Shared.Collections;
+using Robust.Shared.Prototypes;
+
+namespace Content.IntegrationTests.Tests.Roles;
+
+[TestFixture]
+public sealed class LoadoutTests
+{
+    private EntityQuery<StorageComponent> _storageQuery;
+
+    /// <summary>
+    /// Checks that a storage fill on a StartingGearPrototype will properly fill
+    /// </summary>
+    [Test]
+    public async Task TestStartingGearStorage()
+    {
+        var settings = new PoolSettings { Connected = true, Dirty = true };
+        await using var pair = await PoolManager.GetServerClient(settings);
+        var mapManager = pair.Server.ResolveDependency<IMapManager>();
+        var transformSystem = pair.Server.ResolveDependency<SharedTransformSystem>();
+        var storageSystem = pair.Server.ResolveDependency<StorageSystem>();
+        var server = pair.Server;
+        var client = pair.Client;
+
+        Assert.That(server.CfgMan.GetCVar(CVars.NetPVS), Is.False);
+
+        var protoIds = server.ProtoMan
+            .EnumeratePrototypes<StartingGearPrototype>()
+            .Where(p => !p.Abstract)
+            .Where(p => !pair.IsTestPrototype(p))
+            .Select(p => p.ID)
+            .ToList();
+
+        protoIds.Sort();
+        var mapId = MapId.Nullspace;
+
+        await server.WaitPost(() =>
+        {
+            mapId = mapManager.CreateMap();
+        });
+
+        var coords = new MapCoordinates(Vector2.Zero, mapId);
+
+        await pair.RunTicksSync(3);
+
+        foreach (var protoId in protoIds)
+        {
+            server.ProtoMan.TryIndex(protoId, out StartingGearPrototype gearProto);
+
+            var backpackProto = gearProto.GetGear("Back");
+
+            EntityUid bag = default;
```

Comment:
pretty sure you can just set this when you spawn

Reply:
idk what this means

Reply:
i mean that i dont think bag is used in the scope it is on and can be moved to when its spawned (something like var bag = Spawn)

Also ask for reviews again when the tests pass (or when you are stuck and need help)